### ISSUE TITLE
Add focus trap to `ResourceCancelModal` modal component

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#13908]
**Is your feature request related to a problem? Please describe.**

Add focus trap to `ResourceCancelModal` modal component

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by a-arias** [comment]
**TESTS** 
Rancher : v2.12-2d5eaedb623b5b3a1820326808aaa0f8b4d94daa-head
UI:master https://github.com/rancher/dashboard/commit/019581fe6d9794566e252cfd509ff0b3ba6f465f

Fixes verified

**Areas or cases that should be tested**
the assignTo modal needs a feature flag provisioningv2-fleet-workspace-back-population to be enabled first, then an option appears in the Fleet cluster list view for table actions called Change Workspace ✅ 

the CommunityLinks modal only appears in the links in the homepage IF the language is set to Chinese on the user preferences (it will be the last link) ✅ 

The ResourceCancelModal modal appears when editing a resource (edit view) then you click on the option in the footer Edit as YAML, make some change on the YAML, then click on Back to Form ✅ 


<img width="1277" alt="Image" src="https://github.com/user-attachments/assets/ec407a4a-9ba9-407d-b799-dd4d0a1d4a0a" />
<img width="1198" alt="Image" src="https://github.com/user-attachments/assets/792ee6c4-3384-4b9b-a3d7-43eb232618b5" />
<img width="1147" alt="Image" src="https://github.com/user-attachments/assets/a26d3cb5-b030-4bc6-8f21-7c04ffeeb3a0" />
